File: U2-Toolkit/.NET Samples/samples/VB.NET/UniData/EntityFramework_CodeFirst/Student.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.ComponentModel.DataAnnotations
Imports System.Data.Entity
Imports System.Data.Entity.Infrastructure

Namespace EntityFramework_CodeFirst
    <Table("STUDENT_NF_SUB", Schema:="ppp")> _
    Public Class Student
        Sub New()
            Semesters = New HashSet(Of Semester)()
        End Sub

        <Key()> _
        Public Property ID() As String

        Public Property FNAME() As String

        Public Overridable Property Semesters() As ICollection(Of Semester)


    End Class

    <Table("STUDENT_CGA_MV_SUB", Schema:="ppp")> _
    Public Class Semester
        Sub New()
            Courses = New HashSet(Of Course)()
        End Sub


        Public Property ID() As String

        <Key()> _
        Public Property CGA_MV_KEY() As Integer

        Public Property SEMESTER() As String

        Public Overridable Property Student() As Student
        Public Overridable Property Courses() As ICollection(Of Course)

    End Class


    <Table("STUDENT_CGA_MS_SUB", Schema:="ppp")> _
    Public Class Course

        Sub New()
        End Sub


        Public Property ID() As String


        Public Property CGA_MV_KEY() As Integer

        <Key()> _
        Public Property CGA_MS_KEY() As Integer

        Public Property COURSE_NAME() As String

        Public Property TEACHER() As String

        Public Overridable Property Semester() As Semester


    End Class

    Public Class StudentContext
        Inherits DbContext

        Public Sub New()
        End Sub

        Public Property Students() As DbSet(Of Student)
        Public Property Semesters() As DbSet(Of Semester)
        Public Property Courses() As DbSet(Of Course)

        Protected Overrides Sub OnModelCreating(modelBuilder As DbModelBuilder)

            modelBuilder.Conventions.Remove(Of IncludeMetadataConvention)()
            ' this will take care of 
            'SELECT Extent1.Id, Extent1.ModelHash FROM dbo.EdmMetadata AS Extent1 ORDER BY Extent1.Id DESC SAMPLE 1 


            modelBuilder.Entity(Of Student)().[Property](Function(s) s.ID).IsRequired()
        End Sub


    End Class
End Namespace

File: U2-Toolkit/.NET Samples/samples/C#/UniData/EntityFramework_CodeFirst/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel.DataAnnotations;
using System.Data.Entity;
using System.Data.Entity.Infrastructure;

namespace EntityFramework_CodeFirst
{
    [Table("STUDENT_NF_SUB", Schema = "ppp")]
    public class Student
    {
        Student()
        {
            Semesters = new HashSet<Semester>();
        }
        [Key]
        public string ID { get; set; }

        public string FNAME { get; set; }

        public virtual ICollection<Semester> Semesters { get; set; }

        
    }

    [Table("STUDENT_CGA_MV_SUB", Schema = "ppp")]
    public class Semester
    {
        Semester()
        {
            Courses = new HashSet<Course>();
        }
       
        public string ID { get; set; }

        [Key]
        public int  CGA_MV_KEY { get; set; }

        public string SEMESTER { get; set; }

        public virtual Student Student { get; set; }
        public virtual ICollection<Course> Courses { get; set; }

    }


    [Table("STUDENT_CGA_MS_SUB", Schema = "ppp")]
    public class Course
    {
        Course()
        {

        }

        public string ID { get; set; }

        
        public int CGA_MV_KEY { get; set; }

        [Key]
        public int CGA_MS_KEY { get; set; }

        public string COURSE_NAME { get; set; }

        public string TEACHER { get; set; }

        public virtual Semester Semester { get; set; }


    }

    public class StudentContext : DbContext
    {
        public StudentContext()
        {

        }
        public DbSet<Student> Students { get; set; }
        public DbSet<Semester> Semesters { get; set; }
        public DbSet<Course> Courses { get; set; }
        
        protected override void OnModelCreating(DbModelBuilder modelBuilder)
        {
        
            modelBuilder.Conventions.Remove<IncludeMetadataConvention>(); 
            // this will take care of 
            //SELECT Extent1.Id, Extent1.ModelHash FROM dbo.EdmMetadata AS Extent1 ORDER BY Extent1.Id DESC SAMPLE 1 

            modelBuilder.Entity<Student>()
                .Property(s => s.ID)

                .IsRequired();
        }

    }
}
